test/lit/passes/Oz.wast

```diff
@@ -150,4 +154,50 @@
       (i32.const 0x80000000)
     )
   )
+
+  ;; CHECK:      (func $11 (; has Stack IR ;) (param $0 i32) (result i32)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (local.get $0)
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (local.get $0)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (i32.const 0)
+  ;; CHECK-NEXT: )
+  (func $11 (export "eliminate-redundant-checks-1") (param $0 i32) (result i32)
+    (if
+      (local.get $0)
+      (if
+        (local.get $0)
+        (return (local.get $0))
+      )
+    )
+    (i32.const 0)
+  )
+
+  ;; CHECK:      (func $12 (; has Stack IR ;) (param $0 i32) (param $1 i32) (result i32)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (select
+  ;; CHECK-NEXT:    (local.tee $1
+  ;; CHECK-NEXT:     (local.get $0)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (i32.const 0)
+  ;; CHECK-NEXT:    (local.get $1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (local.get $1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (i32.const 0)
+  ;; CHECK-NEXT: )
+  (func $12 (export "eliminate-redundant-checks-2") (param $0 i32) (param $1 i32) (result i32)
+    (if
+      (local.tee $1 (local.get $0))
+      (if
+        (local.get $1)
+        (return (local.get $1))
+      )
+    )
+    (i32.const 0)
+  )
```

Comment:
@kripken main problem is how efficiently handle such cases when we have 'tee' insted simple `get`? Hmm

Reply:
This case is the most common in our case, and we would like to reduce it to:
```wat
  (if
    (local.tee $1 (local.get $0))
    (return (local.get $1))
  )
  (i32.const 0)
```

Reply:
I currently solved it in this way: 1337af9ca5a6c6bb27f8b650f568810da7a8e722 but not sure is it best solution.